plugin/postprocess.py

```diff
@@ -129,47 +132,100 @@ def postprocess_site(
         print(f"No HTML files found in {site_dir}")
         return
 
+    worker_count = os.cpu_count() or 1
+
     # Build markdown index once (O(N) instead of O(N²)) using relative paths as keys
     md_index = {}
     if docs_dir.exists():
         for md_file in docs_dir.rglob("*.md"):
             rel_path = md_file.relative_to(docs_dir).with_suffix("").as_posix()
             md_index[rel_path] = str(md_file)
 
-    print(f"Processing {len(html_files)} HTML files in {site_dir}")
+    mode = "process" if use_processes else "thread"
+    print(f"Processing {len(html_files)} HTML files in {site_dir} with {worker_count} {mode} worker(s)")
 
     processed = 0
     repo_url = None
     git_data = None
     if (add_authors or add_json_ld) and md_index:
         repo_url, git_data = processor.build_git_map(list(md_index.values()))
 
-    progress = TQDM(html_files, desc="Postprocessing", unit="file", disable=not verbose) if TQDM else None
-    log_fn = (progress.write if verbose and progress else print) if verbose else None
-    iterator = progress if progress else html_files
-    for html_file in iterator:
-        success = process_html_file(
-            html_file,
-            site_dir,
-            md_index,
-            git_data,
-            repo_url,
-            site_url=site_url,
-            default_image=default_image,
-            default_author=default_author,
-            add_desc=add_desc,
-            add_image=add_image,
-            add_keywords=add_keywords,
-            add_share_buttons=add_share_buttons,
-            add_authors=add_authors,
-            add_json_ld=add_json_ld,
-            add_css=add_css,
-            add_copy_llm=add_copy_llm,
-            verbose=verbose,
-            log=log_fn,
-        )
-        if success:
-            processed += 1
+    progress = TQDM(total=len(html_files), desc="Postprocessing", unit="file", disable=not verbose) if TQDM else None
+    # For process pools, use a simple print function to avoid pickle issues with bound methods
+    log_fn = None
+    if verbose:
+        if use_processes:
+            log_fn = print
```

Comment:
❗ **CRITICAL**: Passing `print` into `process_html_file` when `use_processes=True` causes every `ProcessPoolExecutor` task to fail before it even runs, because built-in functions are not pickleable. As soon as you try to submit the work you’ll get `TypeError: cannot pickle builtin_function_or_method`, so default process-based execution is currently broken.

**Suggested change:**
```suggestion
            log_fn = None
```